datadayessentials/config/_config.py

```diff
@@ -1,233 +1,161 @@
-            ["azure", "environments", environment_name]
-        )
+    def set_environment_variables(self):
+        """
+        Sets environment variables based on the cloud configuration.
+        This method fetches values from Azure App Configuration and sets environment variables accordingly.
+        If in production, It will fetch values from Azure App Configuration and set environment variables accordingly.
+        If in development, it will fetch values from Azure App Configuration using manual authentication
+            and set environment variables accordingly.
+        """
+        if self.in_production_environment():
+            # TODO: Implement getting connection string in GitHub actions
+            self.create_azure_app_configuration_client_connection("placeholder")
+            self.set_environment_variables_from_cloud_config()
 
-    @staticmethod
-    @catch_key_error
-    def get_machine_learning_workspace():
-        machine_learning_workspace_ref = LocalConfig.get_value_from_config(
-            [
-                "azure",
-                "environments",
-                LocalConfig.ENVIRONMENT,
-                "machine_learning_workspace",
-            ]
-        )
-        return LocalConfig.get_value_from_config(
-            ["azure", "machine_learning_workspaces", machine_learning_workspace_ref]
-        )
+        if not self.in_production_environment():
+            self.client_azure_connection()
+            self.set_environment_variables_from_cloud_config()
 
-    @staticmethod
-    @catch_key_error
-    def get_database_credentials(credentials_reference: str) -> dict:
-        key_vault = LocalConfig.get_environment()["key_vault"]
-        return LocalConfig.get_value_from_config(
-            [
-                "azure",
-                "key_vaults",
-                key_vault,
-                "database_credentials",
-                credentials_reference,
-            ]
-        )
+    def create_azure_app_configuration_client_connection(self, connection_string: str):
+        """
+        Creates a connection to Azure App Configuration.
 
-    @staticmethod
-    @catch_key_error
-    def get_database(database_reference: str) -> dict:
-        return LocalConfig.get_value_from_config(["databases", database_reference])
+        Args:
+            connection_string (str): The connection string for the Azure App Configuration.
 
-    @staticmethod
-    @catch_key_error
-    def list_available_databases() -> list:
-        return list(LocalConfig.get_value_from_config(["databases"]).keys())
+        Raises:
+            ConnectionError: If an error occurs while connecting to Azure.
+        """
+        try:
+            credential = DefaultAzureCredential()
+            credential = DataLakeAuthentication()
+            self.azure_app_config_connection_client = (
+                AzureAppConfigurationClient.from_connection_string(connection_string)
+            )
+        except Exception as e:
```

Comment:
This is the right try - catch approach 👍  I would suggest including the error that you want to catch rather than a generic Exception as this could lead to unintended silent errors.

Reply:
Might be worth creating a new Authentication method in essentials.. DataLakeAuthentication is too specific here.  Implementation is very similar to DLAuthentication, but will improve readability.  You will need to check first if os.environ["CONFIG_CONNECTION_STRING"] exists, if so, use that, otherwise, call the credential=DataLakeAuthentication (or equivalent) and pass that to AzureAppConfigurationClient.  You wont need both DefaultAzureCredential and the DLAuthentication

Reply:
Think this still needs the specific error that you are trying to catch